client/src/components/ElementEditor/AddElementPopoverContent.js

```diff
@@ -0,0 +1,58 @@
+/* global window */
+
+import React, { Component, PropTypes } from 'react';
+import { Button } from 'reactstrap';
+import classNames from 'classnames';
+
+/**
+ * The AddElementPopoverContent component used in the context of an ElementEditor shows the
+ * available elements that can be added to an ElementalArea.
+ */
+class AddElementPopoverContent extends Component {
+  constructor(props) {
+    super(props);
+
+    this.renderElementButtons = this.renderElementButtons.bind(this);
+  }
+
+  renderElementButtons() {
+    const { elementTypes } = this.props;
+
+    return elementTypes.map((elementType) =>
+      (
+        <Button className={
+          classNames(
+            elementType.icon,
+            'btn--icon-xl',
+            'element-editor__add-element-popover-content--button'
+          )
+        }
+        >
+          {elementType.title}
+        </Button>
+      )
+    );
+  }
+
+  render() {
+    return (
+      <div className="element-editor__add-element-popover-content">
```

Comment:
You're going to hate me for this one. I think we should refactor these class names. I'm terrible at this myself because [BEM confuses me](http://getbem.com/naming/). It's supposed to be like this:

```
.block__element--modifier
```

Modifiers should only be things like `--hidden` or `--disabled`

We should probably refactor this so `element-editor-add-element` is the "block" and `content` is the "element". Ie. `className="element-editor-add-element__content"`. And then for the next div `className="element-editor-add-element__button-container"`.